datalad/interface/tests/test_utils.py

```diff
@@ -0,0 +1,82 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Test dirty dataset handling
+
+"""
+
+__docformat__ = 'restructuredtext'
+
+from os.path import join as opj
+from nose.tools import assert_raises, assert_equal
+from datalad.tests.utils import with_tempfile, assert_not_equal
+from datalad.interface.utils import handle_dirty_dataset
+from datalad.distribution.dataset import Dataset
+
+_dirty_modes = ('fail', 'ignore', 'save-before')
+
+
+def _check_all_clean(ds, state):
+    assert state is not None
+    for mode in _dirty_modes:
+        # nothing wrong, nothing saved
+        handle_dirty_dataset(ds, mode)
+        assert_equal(state, ds.repo.get_hexsha())
+
+
+def _check_auto_save(ds, orig_state):
+    handle_dirty_dataset(ds, 'ignore')
+    assert_raises(RuntimeError, handle_dirty_dataset, ds, 'fail')
+    handle_dirty_dataset(ds, 'save-before')
+    state = ds.repo.get_hexsha()
+    assert_not_equal(orig_state, state)
```

Comment:
is there a point of passing `orig_state` explicitly instead of just `orig_state = ds.repo.get_hexsha()` on top of the function call?   if no reason - then why bother doing it?
I do not see that state used anywhere in the `test_dirty` itself so theoretically no need even to return anything (I don't mind returning it though)